lib/galaxy/managers/notification.py

```diff
@@ -0,0 +1,100 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from sqlalchemy import false
+
+from galaxy import model
+from galaxy.exceptions import (
+    Conflict,
+    ObjectAttributeMissingException,
+    ObjectNotFound,
+)
+from galaxy.managers.base import decode_id
+from galaxy.managers.context import ProvidesAppContext
+from galaxy.schema.fields import EncodedDatabaseIdField
+from galaxy.structured_app import MinimalManagerApp
+from galaxy.web import url_for
+
+
+class NotificationManager:
+    """Interface/service object shared by controllers for interacting with notifications."""
+
+    def __init__(self, app: MinimalManagerApp) -> None:
+        self._app = app
```

Comment:
You should only need the database session here. If a Manager doesn't need app (a [god-object](https://en.wikipedia.org/wiki/God_object), we want to remove its usage whenever possible) it shouldn't receive it.

Reply:
Should I use MinimalManagerTool instead, it contains a galaxy scoped session.

Reply:
No, you don't need app here at all.